rfcs/text/onChange.md

```diff
@@ -0,0 +1,128 @@
+- Start Date: 2019-10-23
+
+# Summary
+
+`onChange` & `onBlur` interface standardisation
+
+# Basic example
+
+```js
+const internalChange = e => {
+    e.target = {
+      ...(name && { name }),
+      ...(id && { id }),
+      value: [
+        {
+          displayText: e.target.innerText,
+          optionValue: e.target.value
+        }
+      ]
+    };
+
+    onChange(e);
+  };
+```
+
+# Motivation
+
+Currently the `onChange` and `onBlur` handers in Carbon use a mix of custom objects `{value: "my-value"}` and React's `SyntheticEvent`.
+
+In the custom objects `name` or `id` are passed inconsistently.
+
+The purpose of this RFC is to standardise the approach so users can rely upon the interface between Carbon their application.
+
+The approach that we choose should be consistent across components so users familiar with the interface of one component can easily use another component.
+
+# Detailed design
+
+Users who implement this should be familiar with React's [SyntheticEvent](https://reactjs.org/docs/events.html).
+
+When writing a Carbon component the user should wrap the user provided `onChange`/`onBlur` callback with code similar to that in
+the basic example.
+
+* Components **MUST** replace `SyntheticEvent.target` to steer users from upon the component implementation details.
+* `SyntheticEvent.target` **MUST** be an object with the properties `name`, `id`, `value`.
+* `SyntheticEvent.target` **MUST** include the `name` and `id` of the component.
+```js
+<Select id="foo" name="foo">
+  <Option id="bar" name="bar" value="bar">Bar</Option>
+</Select>
+```
+> In the example above the `SyntheticEvent.target` **MAY** be the follow `{"id": "foo", "name":"foo", "value": {"optionValue": "bar", "optionText": "Bar"}}`
+
+* `SyntheticEvent.target` **MUST** have a value property, that property can be any value **UNLESS** the component supports
+multiple select, in that case the property must be an `Array` of values. If the component supports both single select
+and multiple select controlled by a prop the property **MUST** be an `Array`.
+* Components **MUST NOT** iterate over or replace props on the original `SyntheticEvent.target`
+* Components **MAY** reference values from the original `SyntheticEvent.target` or use `SyntheticEvent.nativeEvent.target` as they
+are synonymous.
+
+The approach suggested would modify the `SyntheticEvent` object that React's `EventPool` uses for performance optimisation.
+
+When validating this approach it was important to ensure that we didn't adversely affect the `EventPool`.
+
+* `SyntheticEvent` is constructed from a object in the `pool` [SyntheticEvent.js#L94](https://github.com/facebook/react/blob/ab1a4f249e61045d523ddbbfb840e868afbbf785/packages/react-dom/src/events/ChangeEventPlugin.js#L53)
```

Comment:
I think it'd be clearer if we use a tag (`v16.8.2`) instead of a commit hash for all these URLs:
```suggestion
* `SyntheticEvent` is constructed from a object in the `pool` [SyntheticEvent.js#L94](https://github.com/facebook/react/blob/v16.8.2/packages/react-dom/src/events/ChangeEventPlugin.js#L53)
```